rewriter/ChainedSig.cc

```diff
@@ -0,0 +1,88 @@
+#include "rewriter/ChainedSig.h"
+#include "ast/Helpers.h"
+#include "ast/treemap/treemap.h"
+#include "core/errors/infer.h"
+
+using namespace std;
+namespace sorbet::rewriter {
+
+struct ChainedSigWalk {
+    vector<ast::Send *> incompleteSigs;
+
+    ast::ExpressionPtr postTransformSend(core::MutableContext ctx, ast::ExpressionPtr tree) {
+        auto *send = ast::cast_tree<ast::Send>(tree);
+
+        // When we receive a `sig` send, we don't know ahead of time of whether it's a chained sig
+        // or an incomplete `sig` without a block. Save it for later, so that we can add errors
+        // to sigs missing block parameters
+        if (send->fun == core::Names::sig() && send->block == nullptr) {
+            incompleteSigs.push_back(send);
+            return tree;
+        }
```

Comment:
Let me know what you folks think about this approach. I wanted to emit the error of missing block in cases where people do this
```ruby
sig # no block!
def foo; end
```
But I need a way of avoiding emitting this error for chained sigs. For example, in this snippet
```ruby
sig.abstract { void }
```
the walker is invoked both with `sig` alone and with `sig.abstract` as send nodes. In these cases, it's not incorrect to not pass a block to the `sig` method, so we don't want to add errors.